networkx/algorithms/tests/test_swap.py

```diff
@@ -53,6 +71,31 @@ def test_connected_double_edge_swap_star_low_window_threshold():
     assert degrees == sorted(d for n, d in graph.degree())
 
 
+def test_directed_edge_swap_small():
+    with pytest.raises(nx.NetworkXError):
+        G = nx.directed_edge_swap(nx.gnp_random_graph(3, 0.5, directed=True))
```

Comment:
Perhaps better to use a known graph than a random graph. It also makes it easier to see what the function is testing.
```suggestion
        G = nx.directed_edge_swap(nx.path_graph(3, create_using=nx.DiGraph)
```